rust/connlib/tunnel/src/control_protocol/gateway.rs

```diff
@@ -143,4 +94,65 @@ where
             peer.inner.add_resource(resource, expires_at);
         }
     }
+
+    async fn new_tunnel(
+        self: Arc<Self>,
+        peer_config: PeerConfig,
+        client_id: ClientId,
+        resource: ResourceDescription,
+        expires_at: DateTime<Utc>,
+        ice: Arc<RTCIceTransport>,
+    ) -> Result<()> {
+        tracing::trace!(?peer_config.ips, "new_data_channel_open");
+        {
+            let device = self.device.load().clone().ok_or(Error::NoIface)?;
+            for &ip in &peer_config.ips {
+                assert!(device.add_route(ip, self.callbacks()).await?.is_none(),  "gateway does not run on android and thus never produces a new device upon `add_route`");
```

Comment:
Same as for the client, would be great to combine all these `.await`s into a single task and make `new_tunnel` sync.